content/architecture/use-modules-to-define-features.md

```diff
@@ -0,0 +1,114 @@
+---
+title: use modules to define features
+author: 
+  name: Billy Lando
+  url: https://github.com/billyjov
+---
+
+# Problem 
+
+As an app scales, the root module start growing. Loading multiple components increase the start time and affect the application performance.    
+The more components added, the more confusing it will be to maintain the entire application.
+
+# Solution 
+
+Module in Angular is a mechanism to group and encapsulate components, services, directives and pipes.   
+Using module to define features in Angular allows lazy loading, isolation and portability of this features. Feature modules are designed to be used accross multiple components and and their UI will be hosted in a parent Module e.g `AppModule`. They are ways to keep your Angular app clean and streamline your code.    
+Angular allow you to lazy load each feature module in order to boost the performance of your app.
+Feature encapsulation using module allow you to replace each module without (or with just small changes) affect other module. It extend Angular application with capabilities from external libraries and relieved a future inclusion of new concepts like _micro-frontends_ or _monorepos_.
+
+Here's an example. 
+
+```ts
+
+// AddPeopleComponent
```

Comment:
To save some space I would write the decorator in one line:

```
@Component({ ... })
```

I would also remove the comment because it doesn't really add value here.